cmd/kube-apiserver/app/server.go

```diff
@@ -279,104 +192,34 @@ func Run(s *options.APIServer) error {
 		glog.Errorf("Failed to create clientset: %v", err)
 	}
 
-	legacyV1Group, err := registered.Group(api.GroupName)
-	if err != nil {
-		return err
-	}
-
-	storageDestinations := genericapiserver.NewStorageDestinations()
-
-	storageVersions := s.StorageGroupsToGroupVersions()
-	if _, found := storageVersions[legacyV1Group.GroupVersion.Group]; !found {
-		glog.Fatalf("Couldn't find the storage version for group: %q in storageVersions: %v", legacyV1Group.GroupVersion.Group, storageVersions)
-	}
-	etcdStorage, err := newEtcd(api.Codecs, storageVersions[legacyV1Group.GroupVersion.Group], "/__internal", s.EtcdConfig)
-	if err != nil {
-		glog.Fatalf("Invalid storage version or misconfigured etcd: %v", err)
+	resourceEncoding := genericapiserver.NewDefaultResourceEncodingConfig()
+	for group, storageEncodingVersion := range s.StorageGroupsToGroupVersions() {
+		resourceEncoding.SetVersionEncoding(group, storageEncodingVersion, unversioned.GroupVersion{Group: group, Version: runtime.APIVersionInternal})
 	}
-	storageDestinations.AddAPIGroup("", etcdStorage)
```

Comment:
> Again before this change: an explicit AddAPIGroup was required. Now it silently returns

Resource enablement is now solely the realm of `ResourceConfig` now.  This separates the rest of the flow to be configured separately.  I think that "store to default by default" with enumerated exceptions is what most people expect and want.  Realistically, it's what existed, since people cargo-culted and just added the storage.